### PR TITLE
Add ElmanCell and ElmanRNN

### PR DESCRIPTION
**Pull request by anderspapitto**

### CODE REVIEW

File: caffe2/python/rnn_cell.py [hunk 1]
```diff
@@ -234,6 +234,71 @@ def create_states(self, model):
...
+        brew.fc(
+            model,
+            hidden_t_prev,
+            gates_t,
+            dim_in=self.hidden_size,
+            dim_out=self.hidden_size,
+            axis=2,
+        )
+        brew.sum(model, [gates_t, input_t], gates_t)
+        if self.activation == 'tanh':
```
Comment (dzhulgakov) [inline caffe2/python/rnn_cell.py:283]: throw if non of the expected ones? and check in the constructor?

File: caffe2/python/rnn_cell.py [hunk 2]
```diff
@@ -234,6 +234,71 @@ def create_states(self, model):
         ]
 
 
+# based on http://pytorch.org/docs/master/nn.html#torch.nn.RNNCell
+class ElmanCell(RNNCell):
```
Comment (dzhulgakov) [inline caffe2/python/rnn_cell.py]: Hm, I haven't seen it called Elman before - it's usually just "rnn" (which is taken) or "simple" or "basic".

Maybe BasicRNNCell then? (similar to TF)

File: caffe2/python/rnn_cell.py [hunk 3]
```diff
@@ -1405,6 +1470,7 @@ def _LSTM(
 
 
 LSTM = functools.partial(_LSTM, LSTMCell)
```
Comment (dzhulgakov) [inline caffe2/python/rnn_cell.py:1480]: hm, this function is not doing just LSTM any more, maybe we should rename it

File: caffe2/python/rnn_cell.py (hunk 2 above)
Comment (anderspapitto) [reply caffe2/python/rnn_cell.py]: the pytorch docs describe it as "a multi-layer Elman RNN", but yeah I can switch to Basic

File: caffe2/python/rnn_cell.py (hunk 3 above)
Comment (anderspapitto) [reply caffe2/python/rnn_cell.py:1480]: to _RNN?

File: caffe2/python/rnn_cell.py (hunk 1 above)
Comment (anderspapitto) [reply caffe2/python/rnn_cell.py:283]: done

File: caffe2/python/rnn_cell.py (hunk 3 above)
Comment (anderspapitto) [reply caffe2/python/rnn_cell.py]: done

**Pull request closed without merging**